vllm/engine/arg_utils.py

```diff
@@ -1210,6 +1213,54 @@ def create_engine_config(self) -> VllmConfig:
         )
 
 
+@dataclass
+class V1EngineArgs(EngineArgs):
+    """Arguments for vLLM engine v1."""
+
+    # V1's default values that differ from the default values in EngineArgs.
+    # This allows to switch between V1 and V0's default behaviour transparently.
+    enable_prefix_caching: bool = True
+    max_num_seqs: int = 1024
+    max_num_batched_tokens: Optional[int] = None
+
+    @staticmethod
+    def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
+        parser = EngineArgs.add_cli_args(parser)
+        return parser
+
+    def create_engine_config(self,
+                             usage_context: Optional[UsageContext] = None
+                             ) -> VllmConfig:
+        assert (usage_context
+                is not None), "usage_context must be provided for V1EngineArgs"
+
+        if self.max_num_batched_tokens is None:
+            if usage_context == UsageContext.LLM_CLASS:
+                logger.warning("Setting max_num_batched_tokens to 8192 "
+                               "for LLM_CLASS usage context.")
+                self.max_num_batched_tokens = 8192
+            elif usage_context == UsageContext.OPENAI_API_SERVER:
+                logger.warning("Setting max_num_batched_tokens to 2048 "
+                               "for OPENAI_API_SERVER usage context.")
+                self.max_num_batched_tokens = 2048
+
+        engine_config = super().create_engine_config(usage_context)
+
+        # TODO (ywang96): Enable APC by default when VLM supports it.
+        if engine_config.model_config.is_multimodal_model:
+            logger.warning(
+                "Prefix caching is currently not supported for multimodal "
+                "models and has been disabled.")
+            engine_config.cache_config.enable_prefix_caching = False
+        return engine_config
+
+
+if envs.VLLM_USE_V1:
+    # Overwrite EngineArgs to use V1EngineArgs
+    # This has to be done before `AsyncEngineArgs` is imported.
+    EngineArgs = V1EngineArgs  # type: ignore
```

Comment:
a suggested edit for convenience if you were going to accept the other one

```suggestion
    # Overwrite EngineArgs to use EngineArgsV1
    # This has to be done before `AsyncEngineArgs` is imported.
    EngineArgs = EngineArgsV1  # type: ignore
```